### DISCUSSION TITLE
Screwdriver which appears next to your cursor in the sub editor's wiring mode gets spawned as an object on the submarine if you press Q

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#14241] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

What the title says. When you're wiring in the submarine editor, there's a screwdriver object attached to your cursor. Pressing the "toggle entity list" button removes it from your inventory and leaves it as an object where your cursor was.
I kept finding random screwdrivers on my submarines. Now I finally know how that happens.

### Reproduction steps

1. Enter the wiring mode in the submarine editor.
2. Press Q, or whatever your "toggle entity list" button is.
3. Screwdriver which was attached to your cursor appears as a physical object where your cursor was and stays there even when you leave the wiring mode.

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Happens outside single player or multiplayer game modes (e.g. game launches on startup, something broken in the main menu)

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
xd, can't reproduce

**Reply by NotWendy** [reply]
So true. Neither can I, it decides to happen at random once every 10 or so hours I spend editing submarines. A rare, but annoying bug.

**Comment by NotWendy** [comment]
Oh my god, it happens when you press Q while wiring. The "toggle entity list" button. It just removes the screwdriver from your inventory and leaves it as an object where your cursor was.

**Comment by Regalis11** [comment]
I'm afraid I couldn't find a way to reproduce this. Pressing Q while rewiring doesn't seem to do anything at my end: toggling the entity list seems to be disabled when rewiring. :/

**Reply by SomeRandomNoobKekeke** [reply]
@NotWendy do you also have drop item on q? 
![barodev small](https://github.com/user-attachments/assets/595eb142-cd45-4d82-8151-4c7054f7bd65)

@Regalis11 Try pressing drop item key

**Reply by kaedepon** [reply]
Reproduction steps
~~1. grab any item in edit mode~~
2. Press drop item key in wiring mode
**It necessary is only in Step 2.

Assign the drop item key. Default is None

https://github.com/user-attachments/assets/d915e44a-00aa-4db5-b156-54cc4998c358

**Reply by Regalis11** [reply]
Hmm, I'm not sure I'd consider that a bug. If you opt to drop the item, why should it be removed as opposed to being dropped?

**Reply by SomeRandomNoobKekeke** [reply]
You usually don't opt to drop screwdriver in wiring mode, you probably don't even realize that it's real
This may happen unintentionally and will take you months to figure out where all those screwdriver are coming from

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/commit/4f011c59de663d22b2e88069fabd126057d46616

**Reply by NotWendy** [reply]
Thank you. And I'm glad people managed to figure out that the drop key was the culprit.
